include/pros/screen.h

```diff
@@ -0,0 +1,352 @@
+ *  large - Large text font size
+ *  medium_center - Medium centered text
+ *  large_center - Large centered text
+ * 
+ */
+typedef enum {
+    E_TEXT_SMALL = 0, 
+    E_TEXT_MEDIUM,
+    E_TEXT_LARGE,
+    E_TEXT_MEDIUM_CENTER,
+    E_TEXT_LARGE_CENTER
+} text_format_e_t;
+
+/**
+ * Can be used to determine what the current touch status is for the touchscreen.
+ *  
+ *  TOUCH_PRESSED - Last interaction with screen was a quick press (not released yet)
+ *  TOUCH_RELEASED - Last interaction with screen was a release
+ *  TOUCH_HELD - User is holding screen down (not released or quick press)
+ * 
+ */
+
+typedef enum {
+    E_TOUCH_RELEASED = 0,
+    E_TOUCH_PRESSED,
+    E_TOUCH_HELD
+} last_touch_e_t;
+
+#ifdef PROS_USE_SIMPLE_NAMES
+#ifdef __cplusplus
+#define TEXT_SMALL pros::E_TEXT_SMALL
+#define TEXT_MEDIUM pros::E_TEXT_SMALL
+#define TEXT_LARGE pros::E_TEXT_SMALL
+#define TEXT_MEDIUM_CENTER pros::E_TEXT_SMALL
+#define TEXT_LARGE_CENTER pros::E_TEXT_SMALL
+#define TOUCH_RELEASED pros::E_TOUCH_RELEASED
+#define TOUCH_PRESSED pros::E_TOUCH_PRESSED
+#define TOUCH_HELD pros::E_TOUCH_HELD
+#else
+#define SMALL_TXT E_SMALL_TXT
+#define MEDIUM_TXT E_MEDIUM_TXT
+#define LARGE_TXT E_LARGE_TXT
+#define MEDIUM_CENTER_TXT E_MEDIUM_CENTER_TXT
+#define LARGE_CENTER_TXT E_LARGE_CENTER_TXT
+#define TOUCH_RELEASED E_TOUCH_RELEASED
+#define TOUCH_PRESSED E_TOUCH_PRESSED
+#define TOUCH_HELD E_TOUCH_HELD
+#endif
+#endif
+
+#ifdef __cplusplus
+namespace c {
+#endif
+
+/******************************************************************************/
+/**                  Screen Graphical Display Functions                      **/
+/**                                                                          **/
+/**   These functions allow programmers to display shapes on the v5 screen   **/
+/******************************************************************************/
+
+/**
+ * Set the pen color for subsequent graphics operations
+ *
+ * \param color	The pen color to set (it is recommended to use values
+ * 		 from the enum defined in colors.h)
+ */
+void screen_set_pen(uint32_t color);
+
+/**
+ * Set the eraser color for clearing and the current background.
+ *
+ * \param color	The background color to set (it is recommended to use values
+ * 					from the enum defined in colors.h)
+ */
+void screen_set_eraser(uint32_t color);
+
+/**
+ *  Get the current pen color.
+ * 
+ * \return The current pen color in the form of a value from the enum defined in colors.h.
+ */
+uint32_t screen_get_pen(void);
+
+/**
+ * Get the current eraser color.
+ *
+ * \return The current eraser color in the form of a value from the enum defined in colors.h.
+ */
+uint32_t screen_get_eraser(void);
+
+/**
+ * Reset the display to the default black screen
+ */
+void screen_clear(void);
+
+/**
+ * Scroll lines on the display upwards.
+ *
+ * \param start_line	The line from which scrolling will start
+ * \param lines			The number of lines to scroll up
```

Comment:
Is the formatting here just a github diff viewer quirk? Looks like a job for clang-format

Reply:
Yeah idk what's up with that... looks find on my end.

Reply:
Appears to me to be a tabs vs spaces thing, FYI.

Reply:
Do y'all have a clang-format script or CI action set up? that would help catch some of these

Reply:
we have a clang-format file locally, I was at one point working on purduesigbots/clang-format-action but other things took priority

Reply:
Turns out to be a tabs vs spaces thing as nick mentioned :/